arc/cli/commands/run.py

```
@@ -416,6 +475,336 @@ def _check_modal_available() -> bool:
+async def _execute_modal_with_streaming(
+    scenarios: List[Scenario],
+    agent_config: Dict[str, Any],
+    agent_profile: Dict[str, Any],
+    funnel_analyzer: FunnelAnalyzer,
+    assumption_detector: AssumptionDetector,
+    execution_loader: ExecutionProgressLoader,
+    json_output: bool
+) -> tuple[List[Dict[str, Any]], float, float]:
+    """Execute Modal scenarios with real-time streaming intelligence."""
+    try:
+        import modal
```

Ruff (0.11.9)

`modal` imported but unused; consider using `importlib.util.find_spec` to test for availability

(F401)